.github/workflows/ci-pull-requests.yml

```diff
@@ -48,7 +48,7 @@ jobs:
       - uses: actions/checkout@v3.5.3
         with:
           repository: n8n-io/n8n
-          ref: ${{ inputs.branch }}
+          ref: refs/pull/${{ github.event.pull_request.number }}/merge
```

Comment:
Note that, since `inputs.branch` is seemingly always empty (as I understand, there are no `inputs` for jobs triggered automatically), the original version passed an empty `ref` argument to `actions/checkout`; in this case, `actions/checkout` [falls back](https://github.com/actions/checkout/blob/v3.6.0/src/input-helper.ts#L60-L64) to `github.context.ref`, which is defined as [PR merge branch](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) already.

So both old and new versions of line 51 have the same effect, and removing line 51 entirely would have the same effect too.

(The same applies to line 13, and to line 40 as well, except that `units-tests-reusable.yml`, as I understand, uses `'master'` as the default value if `ref` is not passed).

The original issue, #7423, has this explained in more detail.

Reply:
Hey @inga-lovinde thanks much for the thorough write up in the issue! I think you are right and I also saw the fallback gives the same value when testing this but decided to keep an explicit `ref` since it is easier to see what's going on like this. Hopefully it will help us with debugging in the future without introducing more edgecases.